src/dpp/channel.cpp

```diff
@@ -478,14 +478,34 @@ channel& channel::fill_from_json(json* j) {
 }
 
 std::string thread::build_json(bool with_id) const {
-	json j = json::parse(channel::build_json(with_id));
-	j["type"] = (flags & CHANNEL_TYPE_MASK);
-	j["thread_metadata"] = this->metadata;
-	if (!this->applied_tags.empty()) {
-		j["applied_tags"] = json::array();
-		for (auto &tag_id: this->applied_tags) {
-			if (tag_id) {
-				j["applied_tags"].push_back(tag_id);
+	json j;
+	j["name"] = this->name;
+	j["archived"] = this->metadata.archived;
+	j["auto_archive_duration"] = this->metadata.auto_archive_duration;
+	j["locked"] = this->metadata.locked;
+	
+	if(is_private_thread()) {
+		j["invitable"] = this->metadata.invitable;
+	}
+
+	j["rate_limit_per_user"] = this->rate_limit_per_user;
+
+	if (is_forum() || is_media_channel()) {
```

Comment:
is_forum or is_media_channel are set on the parent, are they not? not the thread within it?